### PR TITLE
Instantiate the app of apps inside the argocd-helm module

### PR DESCRIPTION
**Pull request by raphink**
This puts the app of apps helm release inside the argocd-helm module, and refactors the variables for more clarity.

It supersedes #340

### CODE REVIEW

**Comment by raphink** [comment]
Tests on https://katacoda.com/raphink/scenarios/devops-stack successful

**Comment by raphink** [comment]
@mcanevet Tests with EKS module successful (training stack).

**Comment by raphink** [comment]
@mcanevet could you test the other stacks (in particular Azure)?

File: modules/aks-azure/main.tf [hunk 1]
```diff
@@ -70,53 +70,33 @@ module "cluster" {
...
+  oauth2_oauth_url                = format("https://login.microsoftonline.com/%s/oauth2/authorize", data.azurerm_client_config.current.tenant_id)
+  oauth2_token_url                = format("https://login.microsoftonline.com/%s/oauth2/token", data.azurerm_client_config.current.tenant_id)
+  oauth2_api_url                  = format("https://graph.microsoft.com/oidc/userinfo")
+  client_id                       = azuread_application.oauth2_apps.application_id
+  client_secret                   = azuread_application_password.oauth2_apps.value
+  cookie_secret                   = random_password.oauth2_cookie_secret.result
+  admin_password                  = ""
+  minio_access_key                = ""
+  minio_secret_key                = ""
+  loki_bucket_name                = "",
```
Comment (mcanevet) [inline modules/aks-azure/main.tf:90]: Triggers this error:
```
Error: Unexpected comma after argument
On .terraform/modules/cluster/modules/aks-azure/main.tf line 90: Argument
definitions must be separated by newlines, not commas. An argument definition
must end with a newline.
```

**Review by mcanevet — CHANGES_REQUESTED** [review]

**Review by mcanevet — APPROVED** [review]

**Pull request merged**